src/parser/FunctionParser.mts

```diff
@@ -222,18 +227,18 @@ export class FunctionParser extends IdentifierParser {
     }
   }
 
-  parseArrowFunction(node, { arrowInfo, Arguments }, kind) {
+  parseArrowFunction(node: ParseNode, { arrowInfo, Arguments }: ParseNode | { arrowInfo?: ArrowInfoStack, Arguments: readonly ParseNode[] }, kind: FunctionKind) {
     const isAsync = kind === FunctionKind.ASYNC;
     this.expect(Token.ARROW);
     if (arrowInfo) {
-      arrowInfo.awaitExpressions.forEach((e) => {
+      arrowInfo.awaitExpressions.forEach((e: ParseNode) => {
```

Comment:
are all these type annotations needed? `arrowInfo.whatever` should be typed already right?

Reply:
because arrowInfo also gets the type from `ParseNode`'s `[index: string]: any` signature. It might be improved in the future when ParseNode is detailed typed.